ciri/node.c

```diff
@@ -90,26 +107,33 @@ retcode_t node_stop(node_t* const node) {
   if (node == NULL) {
     return RC_NODE_NULL_NODE;
   }
+
   log_info(NODE_LOGGER_ID, "Stopping broadcaster component\n");
   if (broadcaster_stop(&node->broadcaster) == false) {
     log_error(NODE_LOGGER_ID, "Stopping broadcaster component failed\n");
     ret = RC_NODE_FAILED_BROADCASTER_STOP;
   }
+
   log_info(NODE_LOGGER_ID, "Stopping processor component\n");
   if (processor_stop(&node->processor) == false) {
     log_error(NODE_LOGGER_ID, "Stopping processor component failed\n");
     ret = RC_NODE_FAILED_PROCESSOR_STOP;
   }
+
   log_info(NODE_LOGGER_ID, "Stopping receiver component\n");
   if (receiver_stop(&node->receiver) == false) {
     log_error(NODE_LOGGER_ID, "Stopping receiver component failed\n");
     ret = RC_NODE_FAILED_RECEIVER_STOP;
   }
+
   log_info(NODE_LOGGER_ID, "Stopping responder component\n");
   if (responder_stop(&node->responder) == false) {
     log_error(NODE_LOGGER_ID, "Stopping responder component failed\n");
     ret = RC_NODE_FAILED_RESPONDER_STOP;
   }
+
+  node->running = false;
```

Comment:
You probably want to set this before entering the stop routines to avoid possible races?

Reply:
If I set it before, we can actually destroy the node with the components still running ?